core/chains/evm/chain_collection.go

```diff
@@ -0,0 +1,108 @@
+package evm
+
+import (
+	"math/big"
+
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/service"
+	"github.com/smartcontractkit/chainlink/core/services/keystore"
+	"github.com/smartcontractkit/chainlink/core/services/postgres"
+	"github.com/smartcontractkit/chainlink/core/store/config"
+	"go.uber.org/multierr"
+	"gorm.io/gorm"
+)
+
+var _ ChainCollection = &chainCollection{}
+
+type ChainCollection interface {
+	service.Service
+	Get(id *big.Int) (Chain, error)
+	Default() (Chain, error)
+	Chains() []Chain
+	ChainCount() int
+}
+
+type chainCollection struct {
+	defaultID *big.Int
+	chains    map[string]*chain
+}
+
+func (cll *chainCollection) Start() (err error) {
+	for _, c := range cll.Chains() {
+		err = multierr.Combine(err, c.Start())
+	}
+	logger.Infof("ChainCollection: Started %d chains, default chain ID is %d", len(cll.chains), cll.defaultID)
+	return
+}
+func (cll *chainCollection) Close() (err error) {
+	for _, c := range cll.Chains() {
+		err = multierr.Combine(err, c.Close())
+	}
+	return
+}
+func (cll *chainCollection) Healthy() (err error) {
+	for _, c := range cll.Chains() {
+		err = multierr.Combine(err, c.Healthy())
+	}
+	return
+}
+func (cll *chainCollection) Ready() (err error) {
+	for _, c := range cll.Chains() {
+		err = multierr.Combine(err, c.Ready())
+	}
+	return
+}
+
+func (cll *chainCollection) Get(id *big.Int) (Chain, error) {
+	if id == nil {
+		panic("FIXME: return default?")
```

Comment:
It is not a good practice to have a panic in runtime. Maybe just return an error?